### PR TITLE
Feature/generate identifier #55

### LINKED ISSUE #55: Automatically generated client identifier consist of too many byte
In the Mqtt specification it is stated that a broker _"MUST allow ClientID’s which are between 1 and 23 UTF-8 encoded bytes"_.

Also _"The Server MAY allow ClientID’s that contain more than 23 encoded bytes"_

To support any broker the client identifier should always be generated so that it bytes do not exceed 23 in length.

**Currently** the cli generates a 54-byte string, e. g:
`hmqClientMQTT_5_0-74eca7ba-5bfe-410a-a724-b956011ef66f`

The **createIdentifier** method in ConnectCommand should be changed so that a 23-byte string is created as an unique identifier.

Also the broker must only accept the following characters in the identfier `"0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"`. Therefore also the creation has to be changed so that only these characters are used.

http://docs.oasis-open.org/mqtt/mqtt/v5.0/cs02/mqtt-v5.0-cs02.html#_Toc514345331

### PR DESCRIPTION
**Pull request by gitseti**
**Motivation**
resolves #55 

**Changes**
* Create a `mqttClient + - + random8CharString` by default as client identifier
* Read in the client length to generate a random client id from properties file
* Cleanup gradle build regarding tests

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -160,19 +160,14 @@ dependencies {
 
     compileOnly 'com.oracle.substratevm:svm:19.2.0'
 
-    testCompile group: 'junit', name: 'junit', version: '4.12'
-    
-    def junitVintageVersion = '5.2.0'
     def junitJupiterVersion = '5.5.1'
     def junitJupiterParamVersion = '5.4.2'
```
Comment (SgtSilvio) [inline build.gradle]: you should use the same version here 5.5.1

File: src/main/java/com/hivemq/cli/DefaultProperties.java [hunk 2]
```diff
@@ -35,12 +35,14 @@
 
     public static final PropertiesUtils.DEBUG_LEVEL SHELL_DEBUG_LEVEL = PropertiesUtils.DEBUG_LEVEL.VERBOSE;
 
-    public static final String CLIENT_PREFIX = "mqttClient";
-
     public static final String LOGFILE_PATH = getLogfilePath();
 
     public static final String SUBSCRIBE_OUTPUT_FILE = null;
 
+    public static final String CLIENT_ID_PREFIX = "mqttClient";
```
Comment (SgtSilvio) [inline src/main/java/com/hivemq/cli/DefaultProperties.java]: Shouldn't this be mqttCli?

File: src/main/java/com/hivemq/cli/utils/MqttUtils.java [hunk 3]
```diff
@@ -65,4 +74,52 @@
                     ", host='" + host + '\'' +
                     '}';
         }
+
+    // See http://docs.oasis-open.org/mqtt/mqtt/v5.0/cs02/mqtt-v5.0-cs02.html#_Toc514345331
+    public static @NotNull String buildRandomClientID(final int length) {
+        if (length < 0) {
+            throw new NegativeArraySizeException("Length of random client id has to be positive");
```
Comment (SgtSilvio) [inline src/main/java/com/hivemq/cli/utils/MqttUtils.java]: IllegalArgumentException would be better imho, you are not creating an array here

File: build.gradle (hunk 1 above)
Comment (gitseti) [reply build.gradle]: resolved in #98

**Pull request merged**